### ISSUE TITLE
Possible compatibility issue with reccomplex

### ISSUE THREAD

**Issue by triem** [srwaggon/minecraft-roguelike#13]
```
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]: java.lang.NullPointerException
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.tasks.DungeonTaskLayout.execute(DungeonTaskLayout.java:31)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.Dungeon.lambda$generate$4(Dungeon.java:210)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:272)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.Dungeon.generate(Dungeon.java:210)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.Dungeon.lambda$null$5(Dungeon.java:225)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.Optional.ifPresent(Optional.java:159)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.Dungeon.lambda$spawnInChunk$6(Dungeon.java:224)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.Optional.ifPresent(Optional.java:159)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.dungeon.Dungeon.spawnInChunk(Dungeon.java:222)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at greymerk.roguelike.DungeonGenerator.generate(DungeonGenerator.java:21)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraftforge.fml.common.registry.GameRegistry.generateWorld(GameRegistry.java:167)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1020)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:999)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.World.func_72964_e(World.java:310)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.World.func_175726_f(World.java:305)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.World.func_180495_p(World.java:911)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.ivtoolkit.world.MockWorld$Real.func_180495_p(MockWorld.java:141)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.expression.PositionedBlockExpression$Argument.at(PositionedBlockExpression.java:67)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.expression.PositionedBlockExpression$Argument.at(PositionedBlockExpression.java:72)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNatural.lambda$canPenetrate$0(TransformerNatural.java:91)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.SupplierCache.get(SupplierCache.java:35)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.expression.PositionedBlockExpression$IsVariableType.lambda$parse$3(PositionedBlockExpression.java:125)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.FunctionExpressionCache.lambda$evaluate$4(FunctionExpressionCache.java:109)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.Algebra$Variable.evaluate(Algebra.java:423)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.BoolAlgebra$3.evaluate(BoolAlgebra.java:48)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.BoolAlgebra$3.evaluate(BoolAlgebra.java:44)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.Algebras$Infix.evaluate(Algebras.java:82)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.Algebra$Operation.evaluate(Algebra.java:496)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.FunctionExpressionCache.evaluate(FunctionExpressionCache.java:109)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.utils.algebra.FunctionExpressionCache.evaluate(FunctionExpressionCache.java:114)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerNatural.canPenetrate(TransformerNatural.java:91)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerAbstractCloud.lambda$buildCloud$1(TransformerAbstractCloud.java:134)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerAbstractCloud.visitRecursively(TransformerAbstractCloud.java:55)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerAbstractCloud.buildCloud(TransformerAbstractCloud.java:119)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerAbstractCloud.configureInstanceData(TransformerAbstractCloud.java:162)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerAbstractCloud.configureInstanceData(TransformerAbstractCloud.java:38)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$configureInstanceData$3(TransformerMulti.java:153)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.ArrayList.forEach(ArrayList.java:1257)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.configureInstanceData(TransformerMulti.java:153)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.configureInstanceData(TransformerMulti.java:41)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.lambda$configureInstanceData$3(TransformerMulti.java:153)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.ArrayList.forEach(ArrayList.java:1257)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.transformers.TransformerMulti.configureInstanceData(TransformerMulti.java:153)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.prepareInstanceData(GenericStructure.java:312)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.generic.GenericStructure.prepareInstanceData(GenericStructure.java:71)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.StructureGenerator.lambda$instanceData$3(StructureGenerator.java:527)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.util.Optional.flatMap(Optional.java:241)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.StructureGenerator.instanceData(StructureGenerator.java:523)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.villages.GenericVillagePiece.prepare(GenericVillagePiece.java:141)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.villages.GenericVillagePiece.generate(GenericVillagePiece.java:180)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.villages.GenericVillagePiece.func_74875_a(GenericVillagePiece.java:172)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.gen.structure.StructureStart.func_75068_a(StructureStart.java:47)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.gen.structure.MapGenStructure.func_175794_a(MapGenStructure.java:94)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.world.gen.feature.structure.MapGenStructureHook.func_175794_a(MapGenStructureHook.java:96)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at ivorius.reccomplex.events.handlers.RCTerrainGenEventHandler$3.func_175794_a(RCTerrainGenEventHandler.java:220)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at mcjty.lostcities.dimensions.world.LostCityChunkGenerator.func_185931_b(LostCityChunkGenerator.java:547)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:999)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.server.MinecraftServer.func_71222_d(MinecraftServer.java:344)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:314)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:270)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486)
[12:58:06] [Server thread/INFO] [STDERR]: [greymerk.roguelike.dungeon.Dungeon:generate:212]:    at java.lang.Thread.run(Thread.java:748)
```

**Comment by srwaggon** [comment]
@triem Thanks for the find. Do you know which version of the mod you are running, or any details about the dungeon's configuration that was being generated at the time?

**Comment by srwaggon** [comment]
This seems to be a duplicate of #14.

I don't believe Reccomplex is incompatible. Rather, I think it is simply asking for the chunks to be generated, which exposes this issue in Roguelike Dungeons.

Fixed in release [`v2.12.2`](https://github.com/srwaggon/minecraft-roguelike/releases/tag/v2.1.2)
